Archivos modificados / creados que deberías revisar y aceptar

Afmastro opened the pull request:
test_firmas.py — parche temporal para compatibilizar con Pydantic v2 (acepta DummyDocumento).
README.md — instrucciones para tests offline, SQL offline y notas de seguridad.
(Si aplicados antes) Otros artefactos añadidos durante la integración: modelos/crud/utils/router/schemas y archivo SQL offline:
document.py
documents.py
firmas.py
firmas.py
documents.py
documents_migration_offline.sql
(Revisa cada uno si quieres cambiar mensajes/docstrings; técnicamente no son necesarios para que el test offline funcione, pero son parte de la integración.)

Qué ya comprobé
Test unitario offline ejecutado con el venv del proyecto y pasó.
Funciones de firma/verificación (RSA PSS + SHA256) probadas localmente.
No apliqué migraciones a la base de datos real (creé SQL offline en documents_migration_offline.sql).
Qué NO está hecho (decisiones pendientes / recomendaciones)
Estos pasos no son obligatorios para aceptar/pushear, pero los recomiendo antes de usar en staging/producción:

El parche en test_firmas.py es temporal para Pydantic v2; idealmente migrar schemas/models a la API de Pydantic v2 y remover el parche.
No se gestionan claves privadas de forma segura aún: en producción debes usar KMS/HSM o almacenamiento seguro; no generes/retornen claves privadas desde el endpoint.
Si quieres persistir claves públicas o añadir columna/relación, aplica la migración (Alembic) en un entorno controlado (staging) con credenciales correctas.
Revisa las warnings de Pydantic v2 y actualiza Config a ConfigDict / from_attributes donde corresponda.

Afmastro committed:
Archivos modificados / creados que deberías revisar y aceptar
test_firmas.py — parche temporal para compatibilizar con Pydantic v2 (acepta DummyDocumento).
README.md — instrucciones para tests offline, SQL offline y notas de seguridad.
(Si aplicados antes) Otros artefactos añadidos durante la integración: modelos/crud/utils/router/schemas y archivo SQL offline:
document.py
documents.py
firmas.py
firmas.py
documents.py
documents_migration_offline.sql
(Revisa cada uno si quieres cambiar mensajes/docstrings; técnicamente no son necesarios para que el test offline funcione, pero son parte de la integración.)

Qué ya comprobé
Test unitario offline ejecutado con el venv del proyecto y pasó.
Funciones de firma/verificación (RSA PSS + SHA256) probadas localmente.
No apliqué migraciones a la base de datos real (creé SQL offline en documents_migration_offline.sql).
Qué NO está hecho (decisiones pendientes / recomendaciones)
Estos pasos no son obligatorios para aceptar/pushear, pero los recomiendo antes de usar en staging/producción:

El parche en test_firmas.py es temporal para Pydantic v2; idealmente migrar schemas/models a la API de Pydantic v2 y remover el parche.
No se gestionan claves privadas de forma segura aún: en producción debes usar KMS/HSM o almacenamiento seguro; no generes/retornen claves privadas desde el endpoint.
Si quieres persistir claves públicas o añadir columna/relación, aplica la migración (Alembic) en un entorno controlado (staging) con credenciales correctas.
Revisa las warnings de Pydantic v2 y actualiza Config a ConfigDict / from_attributes donde corresponda.

Afmastro reviewed (commented):
creo que me fui a la bosta, gemini me ayudo con las pruebas y algunos detalles

Nahuel-a reviewed (changes requested):
Cambiar comentado en la meet

Afmastro committed:
Remove temporary digital-signature test artifacts

Afmastro committed:
Merge cleanup-firmas: remove temporary digital-signature artifacts

Afmastro committed:
Se testeo con base de datos y funciona
Se borraron todos los archivos extras que se
habian creado para pruebas y se dejo solo lo necesario
para que funcione la firma digital

Afmastro committed:
.

Nahuel-a commented on backend/firmas_digitales.py/api_firmas.py:
Corregir nombre de la carpeta -> Sacar el .py porque puede generar errores.

Nahuel-a commented:
Corregir nombres de las funciones creadas. Estan en español y estamos trabajando con los nombres en ingles
Corregir codigos duplicados y verificar sus funcionalidades. Cuales se usan y cuales no para no generar conflictos

Nahuel-a commented on backend/app/models/document.py:
Tendria que haber un unico modelo de document.
Los dos modelos tienen las mismas variables

Nahuel-a commented on backend/app/utils/firmas.py:
```diff
@@ -0,0 +1,113 @@
+import base64
+
+
+def generar_par_claves(key_size: int = 2048):
```
Estamos trabajando con funciones asincronas. 
En este caso deberia de ser "async def" para seguir la misma logica del proyecto

Nahuel-a commented on backend/app/utils/firmas.py:
```diff
@@ -0,0 +1,113 @@
+from typing import Tuple
+
+
+def generar_par_claves(key_size: int = 2048) -> Tuple[bytes, bytes]:
```
Estoy viendo que desde la linea 66 hasta la 113 tiene codigo repetido. Corroborar que es lo que funciona y que es lo que no funciona para eliminar duplicados

Nahuel-a commented on backend/firmas_digitales.py/models_firmas.py:
Aca hay otro modelo de documento que ya estaba creado en models/document.py

Afmastro committed:
Resumen
Esta rama (FirmasDigitales) contiene la integración de la funcionalidad de firmas digitales (schemas, models, CRUD y utilidades). Al intentar ejecutar o importar el router de firmas (api_firmas.py) se detecta un error de importación/compilación porque los archivos schemas.documents y crud.documents (que definen DocumentsIn, DocumentsOut y DocumentsCrud) se implementaron en otra rama y aún no están presentes en FirmasDigitales.

Objetivo
Traer los cambios ya hechos (schemas y crud de documents) desde la rama donde fueron desarrollados, resolver conflictos y dejar la rama FirmasDigitales en un estado importable y testeable.

Qué falla exactamente (error observado)
api_firmas.py referencia:
DocumentsIn, DocumentsOut (Pydantic schemas)
DocumentsCrud (clase CRUD para documentos)
Al no existir esos símbolos en la rama actual, las importaciones fallan y el router no puede ser usado ni testeado.
Nota adicional: la función verify_signature en firmas.py actualmente requiere la clave pública (public_pem) como argumento; el router la invoca asumiendo una firma con distinto arity. Esto debe unificarse (pasar la public key, o adaptar la utilidad para usar una clave por defecto).

Afmastro committed:
Resumen
Esta rama (FirmasDigitales) contiene la integración de la funcionalidad de firmas digitales (schemas, models, CRUD y utilidades). Al intentar ejecutar o importar el router de firmas (api_firmas.py) se detecta un error de importación/compilación porque los archivos schemas.documents y crud.documents (que definen DocumentsIn, DocumentsOut y DocumentsCrud) se implementaron en otra rama y aún no están presentes en FirmasDigitales.

Objetivo
Traer los cambios ya hechos (schemas y crud de documents) desde la rama donde fueron desarrollados, resolver conflictos y dejar la rama FirmasDigitales en un estado importable y testeable.

Qué falla exactamente (error observado)
api_firmas.py referencia:
DocumentsIn, DocumentsOut (Pydantic schemas)
DocumentsCrud (clase CRUD para documentos)
Al no existir esos símbolos en la rama actual, las importaciones fallan y el router no puede ser usado ni testeado.
Nota adicional: la función verify_signature en firmas.py actualmente requiere la clave pública (public_pem) como argumento; el router la invoca asumiendo una firma con distinto arity. Esto debe unificarse (pasar la public key, o adaptar la utilidad para usar una clave por defecto).

Nahuel-a commented on backend/app/alembic/versions/9087a88029a2_add_roles_for_user.py:
Esto no se debe borrar, es lo que conecta la base de datos

Afmastro committed:
No encontre los cambios y los arreglos que me marcaste, no me doy cuenta donde buscar en git hub, esto es todo lo que revise hoy, no lo probe

Afmastro committed:
Esto es lo que integre con lo que habia hecho
esta sin probar, se corrigieron las rutas de los
import para optimizarlas y se agregaron los archivos
faltantes de documentos
